storage/aptosdb/src/state_store/mod.rs

```diff
@@ -267,19 +274,77 @@ impl StateStore {
     pub fn new(
         ledger_db: Arc<DB>,
         state_merkle_db: Arc<DB>,
+        state_kv_db: Arc<DB>,
         state_pruner: StatePrunerManager<StaleNodeIndexSchema>,
         epoch_snapshot_pruner: StatePrunerManager<StaleNodeIndexCrossEpochSchema>,
         buffered_state_target_items: usize,
         max_nodes_per_lru_cache_shard: usize,
         hack_for_tests: bool,
     ) -> Self {
+        if let Some(DbMetadataValue::Version(overall_commit_progress)) = ledger_db
```

Comment:
Would it make sense to disable the truncation part? At the moment it's making my full node fail https://github.com/aptos-labs/aptos-core/issues/7982 

Would this corrupt the state if just comment out this added code?

Reply:
(addressed here https://github.com/aptos-labs/aptos-core/issues/7985)